lib/sync/lock.js

```diff
@@ -1,23 +1,31 @@
-// TODO Use mongodb for locking via https://github.com/chilts/mongodb-lock
-// TODO when using https://github.com/chilts/mongodb-lock, ensure old locks are removed
+var mongoLock = require('@akeating-redhat/mongodb-lock');
```

Comment:
I don't think this will work for node 0.10.

I think you can create a PR to the original module to upgrade the mongodb. Based on @david-martin experience, the author should happy to accept the PR.